stack/base/load-singlesshagent.sh

```diff
@@ -0,0 +1,67 @@
+# Make sure you source this script rather than executing it
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]]
+then
+    echo "You need to source this script, stopping." >&2
+    exit 1
+fi
+
+# run as
+# source load-singlesshagent.sh -v
+# for verbose output
+
+load_singlesshagent() {
+    local VERBOSE
+    local SSH_ENV
+    local SSH_ADD_OUTPUT
+    local SSHADD_RETVAL
+    local NUMKEYS
+
+    VERBOSE=false
+    if [ "$1" == "-v" ]
+    then
+        VERBOSE=true
+    fi
+
+    [ "$VERBOSE" == "true" ] && echo "Single SSH agent script [verbose mode]" >&2
+    SSH_ENV="$HOME/.ssh/agent-environment"
+    # Source SSH settings, if applicable
+    if [ -r "${SSH_ENV}" ]; then
+        source "${SSH_ENV}" 2> /dev/null
+        [ "$VERBOSE" == "true" ] && echo "- sourcing existing environment" >&2
+    else
+        [ "$VERBOSE" == "true" ] && echo "- no existing environment to source" >&2
+    fi
+
+    SSH_ADD_OUTPUT=`ssh-add -l`
```

Comment:
Huh, that is really strange, are you sure? I am only redirecting stderr, that should AFAIK not change the exit code.

Reply:
I am not sure 😛 if I don’t remove it the ssh-agent create clause is not hit. I’ll play more to figure out why. But overall everything works fine no matter if the output is redirected or not.

Reply:
I changed it back to `ssh-add -l 2> /dev/null` and it still works fine. The problem I encountered is not because of this line but the other redirected probably. 

Then this PR is ready, please have another look. @danielhollas